guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+
+from keras_cv.layers import CenterNetLabelEncoder
+from keras_cv.layers import DynamicVoxelization
+
+from keras_cv.models.__internal__.unet import UNet
+
+from keras_cv.models.object_detection_3d.center_pillar import MultiClassDetectionHead
+from keras_cv.models.object_detection_3d.center_pillar import MultiClassHeatmapDecoder
+from keras_cv.models.object_detection_3d.center_pillar import MultiHeadCenterPillar
+
+import tensorflow as tf
+from tensorflow import keras
+
+"""
+3D object detection is the process of identifying, classifying,
+and localizing objects within a 3D space.  Inputs are often in the form of point
+clouds, although 2D images are sometimes used as inputs as well for multi-model
+3D object detection. KerasCV currently supports unimodal point cloud inputs for
+3D object detection.
+
+Point cloud inputs to 3D object detection models typically come from LIDAR
+sensors, and are generally structured as large batches of unsorted points
+captured at a single point in time.
+
+Before we get into the details of KerasCV's 3D object detection offerings, let's
+take a look at a pre-trained KerasCV 3D object detector. This preset has been
+trained on the entirety of the Waymo Open Dataset for 160 epochs, and achieves
+an average precision of 0.7 on the evaluation dataset for both vehicles and
+passengers.
+"""
+
+# TODO(ianstenbit): This doesn't work yet
+# pretrained_model = MultiHeadCenterPillar.from_preset("waymo_open_dataset_vehicles_and_pedestrians")
+# TODO(ianstenbit): Load up an eval frame and run predictions
+
+"""
+Great! Below we show a visualization of what some of these predictions look like
+alongside a rendering of the input point clouds.
+"""
+
+# TODO(ianstenbit): Include a link to a rendering of a visualization.
+
+"""
+Awesome! Now let's check out how to train this sort of model ourselves! First
+let's discuss how data is formatted for 3D object detection in KerasCV.
+In KerasCV, we adopt a data format where point clouds are represented as a
+dictionary with the following structure:
+
+```python
+point_cloud = {
+  "point_xyz": FloatTensor[batch_size, 3]
+  "point_features": FloatTensor[batch_size, num_features]
+  "point_mask": BooleanTensor[batch_size]
+}
+```
+
+The `point_xyz` field represents the XYZ coordinates of each point in the
+point cloud.
+
+The `point_features` field represents the LIDAR features of each point in the
+poin cloud. Typical features include range, intensity, and elongation.
+
+In KerasCV, 3D box targets for object detection are represented as vertical
+pillars rotated with respect to the Z axis. We encode each box as a list (or
+Tensor) of 7 floats: the X, Y, and Z coordinates of the box's center, the width,
+length, and height of the box, and the rotation of the box with respect to the
+Z axis. (This rotation is referrred to as `phi` and is always in radians).
+
+KerasCV's first 3D object detection model offering is a center-based model like
```

Comment:
is this the state of the art model?  If so, maybe provide some indication that this is a powerful model!